src/Microsoft.DotNet.Interactive/Commands/KernelCommand.cs

```diff
@@ -35,11 +35,11 @@ public KernelCommand Parent
         internal set
         {
             _parent = value;
-            var currentSlip = RoutingSlip;
+            var currentSlip = RoutingSlip ?? new RoutingSlip();
             RoutingSlip = new RoutingSlip(_parent?.RoutingSlip);
             foreach (var uri in currentSlip)
             {
-                RoutingSlip.TryAdd(uri);
+                RoutingSlip.TryMarkArrival(uri);
```

Comment:
This doesn't make sense. What if the `Uri` in the source slip was also already completed?

As an aside, there are other potential bugs in this logic. Is setting `Parent` to null a scenario we want to allow? Is reparenting a scenario we want to allow? If not, this should probably no longer be a property setter.